### PR TITLE
Fix for #97 Changes to make hprof dump compatible to python3 while maintaining compatibility with python 2*

### PR DESCRIPTION
**Pull request by bashrc-real**
Few fixes:
1-import functools to support reduce
2. read returns bytes instead of string in python3.Made reduce work with bytes
3. replaced deprecated iter\* with the recommended replacements
4. workaround to generate array in a compatible way in python2 and python3.

### CODE REVIEW

**Comment by onomatopoeia** [comment]
Testing the PR I'm seeing an error when running against a heap dump from open street maps using python 3. Investigating...

File: tools/hprof/dump_classes_from_hprof.py [hunk 1]
```diff
@@ -149,7 +150,11 @@ def parse(self, byte_stream):
             return byte_stream.next_eight_bytes()
         else:
             raise Exception('Invalid HprofBasic type: %s' % self)
-
+def getArray(value):
```
Comment (bertmaher) [inline tools/hprof/dump_classes_from_hprof.py]: nit: whitespace before & after function definitions.

File: tools/hprof/dump_classes_from_hprof.py [hunk 2]
```diff
@@ -149,7 +150,11 @@ def parse(self, byte_stream):
...
+    try:
+        return array(value)
+    except TypeError:
+        return array(str(value)[0])
```
Comment (bertmaher) [inline tools/hprof/dump_classes_from_hprof.py]: Whitespace after this line.

File: tools/hprof/dump_classes_from_hprof.py (hunk 1 above)
Comment (bashrc-real) [inline tools/hprof/dump_classes_from_hprof.py]: Can you provide the particular hprof file? I can try it out too.

**Comment by bashrc-real** [comment]
@onomatopoeia  Can you share the hprof file? One of my earlier commits in the PR had a spurious read that was breaking the stream's offset.

**Pull request closed without merging**